commit e2af33d0e541fcae77c7e0d7a151c4b1788e9ccc
Author: James Page <james.page@chainguard.dev>
Date:   Thu Jul 3 14:07:18 2025 +0100

    AWS-LC-FIPS-2.0.0 compat
    
    Align expected version with the behaviour in AWS-LC-FIPS 2.0.0
    which just returns "AWS-LC".

diff --git a/source/shared/lccrypto_common.c b/source/shared/lccrypto_common.c
index 0a927e0..6fe6735 100644
--- a/source/shared/lccrypto_common.c
+++ b/source/shared/lccrypto_common.c
@@ -84,11 +84,8 @@ void aws_validate_libcrypto_linkage(void) {
      * will need to be addressed by increasing buffer size.*/
     char expected_version[64] = {0};
 #if defined(OPENSSL_IS_AWSLC)
-    /* get FIPS mode at runtime because headers don't give any indication of
-     * AWS-LC's FIPSness at aws-c-cal compile time. version number can still be
-     * captured at preprocess/compile time from AWSLC_VERSION_NUMBER_STRING.*/
-    const char *mode = FIPS_mode() ? "AWS-LC FIPS" : "AWS-LC";
-    snprintf(expected_version, sizeof(expected_version), "%s %s", mode, AWSLC_VERSION_NUMBER_STRING);
+    /* NOTE: AWS-LC 2.0.0 just returns AWS-LC at runtime!!! */
+    snprintf(expected_version, sizeof(expected_version), "AWS-LC");
 #elif defined(OPENSSL_IS_BORINGSSL)
     snprintf(expected_version, sizeof(expected_version), "BoringSSL");
 #elif defined(OPENSSL_IS_OPENSSL)
